test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectIdentityFlowTests.cs

```diff
@@ -0,0 +1,640 @@
+        {
+            options.StateDataFormat = new AuthenticationPropertiesFormater();
+            options.Notifications =
+                new OpenIdConnectAuthenticationNotifications
+                {
+                    SecurityTokenValidated = (notification) =>
+                    {
+                        notification.SkipToNextMiddleware();
+                        return Task.FromResult<object>(null);
+                    }
+                };
+        }
+
+        private static void DefaultOptions(OpenIdConnectAuthenticationOptions options)
+        {
+            options.StateDataFormat = new AuthenticationPropertiesFormater();
+        }
+
+        #endregion
+
+        private static TestServer CreateServer(Action<OpenIdConnectAuthenticationOptions> configureOptions, string scheme)
+        {
+            return TestServer.Create(app =>
+            {
+                app.UseOpenIdConnectAuthenticationPublic(configureOptions);
+                app.Use(async (context, next) =>
+                {
+                    await next();
+                });
+            },
+            services =>
+            {
+                services.AddDataProtection();
+            });
+        }
+    }
+
+    /// <summary>
+    /// Extension specifies <see cref="OpenIdConnectAuthenticationMiddlewarePublic"/> as the middleware.
+    /// </summary>
+    public static class OpenIdConnectAuthenticationExtensions
+    {
+        /// <summary>
+        /// Adds the <see cref="OpenIdConnectAuthenticationMiddleware"/> into the ASP.NET runtime.
+        /// </summary>
+        /// <param name="app">The application builder</param>
+        /// <param name="options">Options which control the processing of the OpenIdConnect protocol and token validation.</param>
+        /// <returns>The application builder</returns>
+        public static IApplicationBuilder UseOpenIdConnectAuthenticationPublic(this IApplicationBuilder app, Action<OpenIdConnectAuthenticationOptions> configureOptions)
+        {
+            return app.UseMiddleware<OpenIdConnectAuthenticationMiddlewarePublic>(new ConfigureOptions<OpenIdConnectAuthenticationOptions>(configureOptions));
+        }
+    }
+
+    public class OpenIdConnectAuthenticationContext : IAuthenticateContext
+    {
+        public OpenIdConnectAuthenticationContext(string scheme = null)
+        {
+            AuthenticationScheme = scheme ?? OpenIdConnectAuthenticationDefaults.AuthenticationScheme;
+        }
+
+        public string AuthenticationScheme
+        {
+            get;
+            set;
+        }
+
+        public void Authenticated(ClaimsPrincipal principal, IDictionary<string, string> properties, IDictionary<string, object> description)
+        {
+        }
+
+        public void NotAuthenticated()
+        {
+        }
+    }
+
+    public class OpenIdConnectAuthenticationHandlerPublic : OpenIdConnectAuthenticationHandler
```

Comment:
Good idea.

Sent from my Windows Phone
________________________________
From: Hao Kung<mailto:notifications@github.com>
Sent: ‎3/‎31/‎2015 10:42 AM
To: aspnet/Security<mailto:Security@noreply.github.com>
Cc: BrentSchmaltz<mailto:brentschmaltz@hotmail.com>
Subject: Re: [Security] logging, added tests, fixed code only flow. (#202)

> +        public string AuthenticationScheme
> +        {
> +            get;
> +            set;
> +        }
> +
> +        public void Authenticated(ClaimsPrincipal principal, IDictionary<string, string> properties, IDictionary<string, object> description)
> +        {
> +        }
> +
> +        public void NotAuthenticated()
> +        {
> +        }
> +    }
> +
> +    public class OpenIdConnectAuthenticationHandlerPublic : OpenIdConnectAuthenticationHandler

nit: consider TestOpenIdConnectAuthHandler instead of Public suffix?

---
Reply to this email directly or view it on GitHub:
https://github.com/aspnet/Security/pull/202/files#r27503331